### PR TITLE
config route refactor

### PR DESCRIPTION
**Pull request by abildw**
* updating config route to use swagger controller 

@dalebremner @geoff-reid @brianparry @srinia6 @RackHD/corecommitters

### CODE REVIEW

File: lib/api/2.0/config.js [hunk 1]
```diff
@@ -3,19 +3,19 @@
...
 
 var injector = require('../../../index.js').injector;
+var controller = injector.get('Http.Services.Swagger').controller;
+var config = injector.get('Http.Services.Api.Config');
 
-module.exports = {
-    getConfig: getConfig,
-    patchConfig: patchConfig
-};
+function configGet = controller(function(req) {
```
Comment (zyoung51) [inline lib/api/2.0/config.js]: should be 'var configGet'.  Same comment for the issue below.

File: lib/services/config-api-service.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ function configApiServiceFactory(
 
     ConfigApiService.prototype.configSet = function(req) {
```
Comment (brianparry) [inline lib/services/config-api-service.js]: The service should just take a config object, not a request.

Comment (brianparry) [inline lib/services/config-api-service.js]: Nevermind, this is doing the right thing.  I'd change the name though. `req` implies a request object to me.

**Comment by abildw** [comment]
@brianparry changed function parameter name from "req" to "config" to avoid confusion

**Comment by srinia6** [comment]
+1

**Comment by brianparry** [comment]
:+1:

**Pull request merged**